### ISSUE TITLE
Plotting bad sf data can make R crash

### ISSUE THREAD

**Issue by karldw** [tidyverse/ggplot2#2172]
Providing bad data to `geom_sf` can lead ggplot to crash the current R session.  

The problem is something related to using different projections in the layers (which I realize is improper, but probably shouldn't crash R). It's also somewhat specific to the projections I've chosen; using Lambert azimuthal equal-area (`+proj=laea`) instead of Robinson (`+proj=robin`) doesn't cause a crash.

```r
library(ggplot2)
library(sf)

robin_crs <- "+proj=robin +lon_0=0 +x_0=0 +y_0=0 +ellps=WGS84 +datum=WGS84 +units=m +no_defs"
india_crs <- "+init=epsg:24383"  # India-centered Lambert conformal conic projection

bad_point <-  st_as_sf(data.frame(longitude = 0, latitude = 92),  # bad latitude
  coords = c("longitude", "latitude"), crs = st_crs("+init=epsg:4267"))

plt <- ggplot() +
  geom_sf(data = st_transform(bad_point, crs = india_crs)) +
  geom_sf(data = st_transform(bad_point, crs = robin_crs))
ggplot_build(plt)  # force calculation here (print would work too)

```
Other details: I'm using the latest development version of ggplot2, 2.2.1.9000, commit f1bef79. This crash happens inside and outside RStudio. It's not a caught segfault; it's an immediate crash. Adding `coord_sf()` doesn't help.

Would you like me to contribute a failing test for this?

**Comment by karawoo** [comment]
This causes a segfault on my machine, but either way it seems like a bug.

**Comment by karldw** [comment]
Thanks!  I just reran on a different machine and got a segfault too.

**Comment by mdmadhu** [comment]
I am not so sure that this is related to 'bad' data (e.g., latitude > 90 degrees). I have development builds of ggplot2 (ggplot2, 2.2.1.9000) and the 0.5.1 version of sf, and even code below produces a crash. Have tried with different projections, but no luck. Not sure if this is a ggplot issue or an sf issue. Hence mentioning @edzer also.

`library(ggplot2)`
`library(sf)`
`point <-  st_as_sf(data.frame(longitude = 76, latitude = 12),   coords = c("longitude", "latitude"), crs = st_crs("+init=epsg:4326"))`
`ggplot() + geom_sf(data = point)`

**Comment by mdmadhu** [comment]
Ok, I just installed the latest dev build (0.5-2) of sf, which seems to solve the problem. No crash anymore and the plotting looks fine. Again mentioning @edzer.

**Comment by edzer** [comment]
Thanks, I filed the issue. @mdmadhu : I don't think there was a change from 0.5-1 to devel (0.5-2) that can affect this, did you also do sth else (e.g. binary install 0.5-1, source install 0.5-2)?

**Comment by mdmadhu** [comment]
Thanks, Edzer, you're right. I did a binary install of 0.5-1, but an
install from source for 0.5-2. Curious how that might have caused/cured
this issue.

**Comment by edzer** [comment]
The version of proj.4 linked to, see https://github.com/edzer/sfr/issues/398#issuecomment-311597128

**Comment by rsbivand** [comment]
Right, do run:

```
sf_extSoftVersion()
```

If PROJ.4 is >= 4.9.3, the problem is fixed there, for <= 4.9.2 it will crash. Thanks to Even Rouault (@rouault)!

**Comment by karawoo** [comment]
I'm closing this as it looks like a fixed issue that's not with ggplot2, but let me know if there are still problems.
